src/search/parser_tests.h

```diff
@@ -0,0 +1,210 @@
+        std::cout << indent << "f_eval = " << std::endl;
+        if (f_eval) {
+            f_eval->dump(indent + "    ");
+        } else {
+            std::cout << indent << "    " << "Not specified on command line" << std::endl;
+        }
+        std::cout << indent << "preferred = " << std::endl;
+        for (const auto& h : preferred) {
+            h->dump(indent + "    ");
+        }
+    }
+};
+
+class EagerSearchPlugin : public parser::Plugin<SearchEngine, EagerSearch> {
+public:
+    EagerSearchPlugin()
+        : parser::Plugin<SearchEngine, EagerSearch>("eager", "Eager Search") {;
+        document_synopsis("Eager search synopsis");
+
+        add_option<bool>("reopen_closed", "reopen closed nodes", "false");
+        add_option<std::shared_ptr<Evaluator>>(
+            "f_eval",
+            "set evaluator for jump statistics. "
+            "(Optional; if no evaluator is used, jump statistics will not be displayed.)",
+            parser::ArgumentInfo::NO_DEFAULT);
+        add_list_option<std::shared_ptr<Evaluator>>(
+            "preferred",
+            "use preferred operators of these evaluators", "[]");
+    }
+};
+
+class IteratedSearch : public SearchEngine {
+    std::vector<std::unique_ptr<parser::EnrichedASTNode>> searches;
+public:
+    explicit IteratedSearch(const parser::Options &opts)
+        : SearchEngine(opts),
+          searches() {
+        std::shared_ptr<parser::EnrichedASTNode> searches_AST =
+            opts.get<std::shared_ptr<parser::EnrichedASTNode>>("searches");
+        std::shared_ptr<parser::EnrichedListNode> searches_AST_list =
+                std::dynamic_pointer_cast<parser::EnrichedListNode>(searches_AST);
+        for (const auto & element : searches_AST_list->get_elements()) {
+            searches.push_back(element->clone());
+        }
+    }
+    virtual ~IteratedSearch() = default;
+    void dump(std::string indent) override {
+        std::cout << indent << "IteratedSearch with " << std::endl;
+        indent += "    ";
+        std::cout << indent << "searches = " << std::endl;
+        for (const auto& search_AST : searches) {
+            parser::Any search_any = search_AST->construct();
+            std::shared_ptr<SearchEngine> search = parser::any_cast<std::shared_ptr<SearchEngine>>(search_any);
+            search->dump(indent + "    ");
+        }
+    }
+};
+
+class IteratedSearchPlugin : public parser::Plugin<SearchEngine, IteratedSearch> {
+public:
+    IteratedSearchPlugin()
+        : parser::Plugin<SearchEngine, IteratedSearch>("iterated", "Iterated Search") {;
+        add_option<std::vector<std::shared_ptr<SearchEngine>>>(
+            "searches",
+            "use preferred operators of these evaluators", "[]", parser::Bounds::unlimited(), true);
+    }
+};
+
+class BlindHeuristicPlugin : public parser::Plugin<Evaluator, BlindHeuristic> {
+public:
+    BlindHeuristicPlugin()
+        : parser::Plugin<Evaluator, BlindHeuristic>("blind", "Blind Heuristic") {
```

Comment:
We thought the identifier ("blind") and the user-readable class name ("Blind Heuristic") are the two pieces of information that are required in any case: without the first, we cannot call the plugin; without the second, we cannot generate documentation for it as this name is used in links, etc. Everything else could be left out, if for example the plugin has no options.
In the old code the name "Blind Heuristic" was passed as part of `add_document_synopsis` but we felt that the name is more mandatory than the synopsis and that `add_document_synopsis` doesn't sound like it would set the synopsis as well as the name.

I have nothing against setting the name as a template parameter but doesn't this just shift the string to the left a bit? I don't see the big advantage. And yes, it is a bit weird.

As for leaving out the template parameters, I wasn't aware that this is possible at all but in this situation it doesn't work: when I remove them, I get "expected class name before '(' token".